flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/plan/batch/sql/join/LookupJoinTest.scala

```diff
@@ -68,8 +68,8 @@ class LookupJoinTest extends TableTestBase {
     expectExceptionThrown(
       "SELECT * FROM MyTable AS T RIGHT JOIN temporalTest " +
         "FOR SYSTEM_TIME AS OF T.proctime AS D ON T.a = D.id",
-      "Unsupported join type for semi-join RIGHT",
-      classOf[IllegalArgumentException]
+      null,
+      classOf[AssertionError]
```

Comment:
Why we need to change this?

Reply:
Calcite does not allow FOR SYSTEM_TIME AS OF non-nullable left table field to Right Join.
There should be a exception:
```
java.lang.AssertionError
	at org.apache.calcite.sql2rel.SqlToRelConverter.getCorrelationUse(SqlToRelConverter.java:2517)
	at org.apache.calcite.sql2rel.SqlToRelConverter.createJoin(SqlToRelConverter.java:2426)
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertFrom(SqlToRelConverter.java:2071)
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelectImpl(SqlToRelConverter.java:646)
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertSelect(SqlToRelConverter.java:627)
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQueryRecursive(SqlToRelConverter.java:3100)
	at org.apache.calcite.sql2rel.SqlToRelConverter.convertQuery(SqlToRelConverter.java:563)
	at org.apache.flink.table.calcite.FlinkPlannerImpl.rel(FlinkPlannerImpl.scala:125)
```

Reply:
Do you mean Calcite doesn't allow **nullable** left field to right join? 

However, the T.proctime field is derived from proctime() builtin method, which should be **not null**.

Reply:
Not nullable, Non-nullable
If this field is non-nullable and this is a right join, after join, this field will be nullable.
Calcite does not deal with this correlation, in `getCorrelationUse`, it will check the input field and output field, their nullable are not the same, so there is a AssertionError.
IMO, Calcite lack of consideration in this situation, but our lookup join don't support it too.